crates/web/src/components/text_field.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashSet;
+
+use leptos::{component, create_memo, view, IntoView, MaybeProp, SignalGet, TextProp};
+
+#[derive(Clone, Debug, Default)]
+pub enum TextFieldVariant {
+    #[default]
+    Primary,
+}
+
+#[derive(Clone, Debug, Default)]
+pub enum TextFieldType {
+    #[default]
+    Text,
+    Email,
+    Password,
+    Number,
+    Search,
+    Tel,
+    Url,
+}
+
+#[component]
+pub fn TextField(
+    #[prop(optional, into)] name: TextProp,
+    #[prop(optional, into)] id: TextProp,
+    #[prop(optional, into)] placeholder: TextProp,
+    #[prop(optional, into)] value: TextProp,
+    #[prop(optional, into)] label: TextProp,
+    #[prop(optional, into)] variant: MaybeProp<TextFieldVariant>,
+    #[prop(optional, into)] r#type: TextFieldType,
+    #[prop(optional, into)] disabled: MaybeProp<bool>,
+    #[prop(optional, into)] full_width: MaybeProp<bool>,
+) -> impl IntoView {
+    let class_names = create_memo(move |_| {
+        let mut classes: HashSet<&str> = HashSet::new();
+
+        match variant.get().unwrap_or_default() {
+            TextFieldVariant::Primary => {
+                classes.insert("px-3.5");
+                classes.insert("py-2");
+                classes.insert("rounded");
+                classes.insert("font-semibold");
+                classes.insert("placeholder:text-purple-200");
+                classes.insert("border-2");
+                classes.insert("border-purple-300");
+                classes.insert("text-purple-400");
+                classes.insert("focus:border-purple-400");
+                classes.insert("focus:ring-purple-500/60");
+            }
+        }
+
+        // Default Classes
+
+        if let Some(is_full_width) = full_width.get() {
+            if is_full_width {
+                classes.insert("w-full");
+            }
+        }
+
+        if let Some(is_disabled) = disabled.get() {
+            if is_disabled {
+                classes.insert("opacity-70");
+                classes.insert("!cursor-not-allowed");
+            }
+        }
+
+        classes.into_iter().collect::<Vec<&str>>().join(" ")
+    });
+    view! {
+            <div>
+            <label class="block mb-2 text-sm font-medium text-purple-500" for=id.clone()>{label}</label>
+            <input type=format!("{:?}", r#type).to_lowercase() name=name value=value id=id placeholder=placeholder class=class_names  disabled=disabled />
```

Comment:
Can we have a `Display` trait implementation for `TextFieldType` so we have a explicit value for each type?

I also suggest narrowing down a bit the variants to just:

- Text
- Email
- Password

```rust
            <input type=format!("{:?}", r#type).to_lowercase()
```

Reply:
Okey!! 🚀